addons/switchunits/functions/fnc_handleMapClick.sqf

```diff
@@ -21,16 +21,11 @@
 params ["_faction", "_pos"];
 _faction params ["", "_sides"];
 
-private _sideNearest = [];
+private _nearestObjects = nearestObjects [_pos, ["Man"], 15];
 
-{
-    if ([_x] call FUNC(isValidAi) && (side group _x in _sides)) then {
-        _sideNearest pushBack _x;
-    };
-    nil
-} count (nearestObjects [_pos, ["Man"], 15]);
+private _nearestValidUnit = _nearestObjects findIf {(side group _x in _sides) && {[_x] call FUNC(isValidAi)}};
```

Comment:
maybe `_nearestValidUnitIndex` name would be better

Reply:
That's actually what I was thinking when writing that.. Apparently the thoughts didn't get to my fingers 🤔